src/schema/schema_neo4j_queries.py

```diff
@@ -652,15 +652,16 @@ def link_collection_to_datasets(neo4j_driver, collection_uuid, dataset_uuid_list
 previous_revision_entity_uuid : str
     The uuid of previous revision entity
 """
-def link_entity_to_previous_revision(neo4j_driver, entity_uuid, previous_revision_entity_uuid):
+def link_entity_to_previous_revision(neo4j_driver, entity_uuid, previous_revision_entity_uuids):
     try:
-        with neo4j_driver.session() as session:
-            tx = session.begin_transaction()
+        for previous_uuid in previous_revision_entity_uuids:
+            with neo4j_driver.session() as session:
+                tx = session.begin_transaction()
 
```

Comment:
We should use the for loop inside the transaction of a session, rather than open a new session and new transaction for each iteration.